### DISCUSSION TITLE
How to use Run and Debug options on Rust file?

### DISCUSSION THREAD

**Discussion by danielnieto** [LunarVim/LunarVim#2868] [Q&A]
So I started learning rust, and when I open the project on Lvim I can see the following:

<img width="267" alt="image" src="https://user-images.githubusercontent.com/2120107/181651077-91d2e974-64c9-4693-aa1d-f920e2eddd67.png">

I cannot click on them, just look at them, is there a command to actually use the "run" or "debug" option? Where do these come from, are they from rust-analyzer? or some other plugin Lvim is using when I edit a rust file?

**Comment by rebuilt** [comment]
Try `<Space>la`  as defined [here](https://github.com/LunarVim/LunarVim/blob/rolling/lua/lvim/core/which-key.lua#L156)

**Reply by danielnieto** [reply]
When I press `<Space>la` within the scope of the function, outside, on the same line as the virtual text I get the same:

<img width="430" alt="image" src="https://user-images.githubusercontent.com/2120107/181806777-0db2a0ba-2384-4ad0-87d6-c96262345d74.png">

**Comment by turutupa** [comment]
commenting just in case anyone publishes an answer

**Reply by turutupa** [reply]
I think its `<leader> l l` being <leader> space bar by default

**Comment by BrillianceSummer** [comment]
i have the same confuse

**Comment by LostInTheLogs** [comment]
it's `leader l l` (code lens)

**Reply by weykon** [reply]
I got this following, but I do not know how to do.
```
Language server does not support
 command `rust-analyzer.runSingle`. This command may require a client extension.
```

**Reply by dumbun** [reply]
same issue
<img width="1680" alt="Screenshot 2024-02-05 at 1 36 53 PM" src="https://github.com/LunarVim/LunarVim/assets/113350510/10ac7dfe-a947-4537-abac-dbe4f6ce083f">
